libs/metrics/in_out.py

```diff
@@ -1,25 +1,58 @@
-import os
+import ast
+import csv
 import json
+import os
 import numpy as np
+
+from numpy import linalg as LA
+from collections import deque
 from datetime import date
 from typing import Dict, Iterator, List
+from datetime import datetime
 
 from .base import BaseMetric
 from constants import IN_OUT
 from libs.utils.config import get_source_config_directory
 from libs.utils.utils import validate_file_exists_and_is_not_empty
-from pathlib import Path
 
 class InOutMetric(BaseMetric):
 
-    reports_folder = IN_OUT
-    csv_headers = []
-    entity = "area"
-    live_csv_headers = []
+    metric_name = IN_OUT
+    reports_folder = metric_name
+    csv_headers = ["In", "Out"]
 
     @classmethod
-    def process_csv_row(cls, csv_row: Dict, objects_logs: Dict):
-        raise NotImplementedError("Operation not implemented")
+    def process_path(cls, boundary_line, trajectory_path):
+        """
+        Verify if a trajectory goes over a boundary line
+        Args:
+            Two coordinates [x,y] are in 2-tuples [A,B]
+                Boundaries of the in/out line.
+                If someone crosses the line while having A to their right, they are going in the in direction (entering)
+                Crossing the line while having A to their left means they are going in the out direction (leaving)
+
+            trajectory_path: List of N 2-tuples (x,y)
+            That represents the trajectory of an object.
+
+        Returns:
+            in, out) : tuple
+                 (1, 0) - if the object entered (in)
+                 (0, 1) - if the object left (out)
+                 (0, 0) - if the object didn't cross the boundary.
+        """
+        number_of_cuts = 5
+        if len(trajectory_path) < number_of_cuts:
+            number_of_cuts = len(trajectory_path)
+
+        intersections = [trajectory_path[int(i)] for i in np.linspace(0, len(trajectory_path) - 1, number_of_cuts)]
```

Comment:
Ideally I would use some kind of outlier removal instead of taking every Nth step but maybe that is for a future version

Reply:
Yeah, I discussed it with Pablo and this seemed to be the better option